vertx-grpc-protoc-plugin2/src/main/java/io/vertx/grpc/plugin/protoc/ProtocRequestConverter.java

```diff
@@ -0,0 +1,174 @@
+package io.vertx.grpc.plugin.protoc;
+
+import com.google.api.AnnotationsProto;
+import com.google.api.HttpRule;
+import com.google.common.html.HtmlEscapers;
+import com.google.protobuf.DescriptorProtos;
+import com.google.protobuf.compiler.PluginProtos;
+import io.vertx.grpc.plugin.descriptors.MethodDescriptor;
+import io.vertx.grpc.plugin.descriptors.ServiceDescriptor;
+import io.vertx.grpc.plugin.descriptors.TranscodingDescriptor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ProtocRequestConverter {
+
+  private final ProtobufTypeMapper typeMapper;
+
+  public ProtocRequestConverter(ProtobufTypeMapper typeMapper) {
+    this.typeMapper = typeMapper;
+  }
+
+  public static ProtocRequestConverter create(PluginProtos.CodeGeneratorRequest request) {
+    ProtobufTypeMapper typeMapper = new ProtobufTypeMapper(request.getProtoFileList());
+    return new ProtocRequestConverter(typeMapper);
+  }
+
+  public List<ServiceDescriptor> convertServices(PluginProtos.CodeGeneratorRequest request) {
+    List<ServiceDescriptor> services = new ArrayList<>();
+
+    for (DescriptorProtos.FileDescriptorProto fileProto : request.getProtoFileList()) {
+      if (!request.getFileToGenerateList().contains(fileProto.getName())) {
```

Comment:
filtering should be done by the caller, that is the `VertxGrpcGenerator` should filter the list to be processed by the `ProtocRequestConverter` instead of relying on it to skip the files that should not be generated